docs/usage/configuration/user_authentication/refresh_tokens.md

```diff
@@ -0,0 +1,124 @@
+    token could go undetected for a very long time.
+* Clients need to implement support for refresh tokens in order for them to be a
+  useful mechanism.
+  * It is up to homeserver administrators if they want to issue long-lived access
+    tokens to clients not implementing refresh tokens.
+    * For compatibility, it is likely that they should at least until client support
+      is widespread.
+      * Users with clients that support refresh tokens will still benefit from the
+        added security; it's not possible to downgrade a session to using long-lived
+        access tokens so this effectively gives users the choice.
+    * In a closed environment where all users use known clients, this may not be
+      an issue as the homeserver administrator can know if the clients have refresh
+      token support.
+
+
+## Configuration Guide
+
+The following configuration options, in the `registration` section, are related:
+
+* `session_lifetime`: maximum length of a session, even if it's refreshed.
+  In other words, the client must log in again after this time period.
+  In most cases, this can be unset (infinite) or set to a long time (years or months).
+* `refreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients supporting refresh tokens.
+  This should be short; a good value might be 5 minutes (`5m`).
+* `nonrefreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients which don't support refresh tokens.
+  Make this short if you want to force use of refresh tokens.
+  Make this long if you don't want to inconvenience users of clients which don't
+  support refresh tokens.
+* `refresh_token_lifetime`: lifetime of refresh tokens.
+  Unless you want to log inactive sessions out, it is often fine to use a long
+  value here or even leave it unset (infinite).
+  Beware that making it too short will inconvenience clients that do not connect
+  very often, including mobile clients and clients of infrequent users.
+
+
+### Using refresh token expiry to log out inactive sessions
+
+If you'd like to force sessions to be logged out upon inactivity, you can enable
+refreshable access token expiry and refresh token expiry.
+
+Assuming that clients keep their access token valid whilst they are active, and
+that you set `refresh_token_lifetime` to some value *R* and
+`refreshable_access_token_lifetime` to some value *A* (which is not bigger than *R*),
+then:
+ 1. sessions may become logged out if they are inactive for more than *R - A*.
+    (Explanation: if a client goes offline just before it was about to refresh its
+    access token, then it must come online before the refresh token expires.)
+ 2. sessions will always become logged out if they are inactive for more than *R*.
+    (Explanation: if a client refreshes its access token just before going offline,
+    then returns just before the refresh token expires, it will be able to continue
+    its session.)
+
```

Comment:
I don't follow this. What is the trigger for a session to be logged out? Your refresh token expires without being used?

Reply:
The key thing I didn't understand here is that the start of the inactivity period can be before your access and refresh tokens were generated. I think the example below would be improved by this.

(Is it always the case that the active access and active refresh tokens are generated at the same instant?)

Reply:
> the start of the inactivity period can be before your access and refresh tokens were generated. I think the example below would be improved by this.

I'm confused by this and on first hand it sounds wrong but I think it's more that I've misunderstood.

> (Is it always the case that the active access and active refresh tokens are generated at the same instant?)

Yup, maybe I should actually explain that 'refreshing an access token' means that you exchange your (old) refresh token for a new access token AND a new refresh token at the same time. It's kind of implicit knowledge to me now but I suppose I didn't actually tell you that before :P